### DISCUSSION TITLE
Security Vulnerability CVE-2023-2603 for proxy-init@v2.2.1

### DISCUSSION THREAD

**Discussion by noodle-rubrik** [linkerd/linkerd2#11296] [General] [answered]
Our security scanner is flagging the below issue for the proxy-init@v2.2.1 image:
https://nvd.nist.gov/vuln/detail/CVE-2023-2603

The underlying issue is the libcap library version 2.66. The scanner is claiming that 2.66-r1 is sufficient to remediate the vulnerability, but it claimed 2.66-r0 would be sufficient earlier, before changing its mind. It looks like 2.69 should definitely be a fix, and the others are some sort of backport.

I was able to clone the linkerd2-proxy repo, checkout the proxy-init@v2.2.1 tag, update the Dockerfile to point to alpine:3.18, and successfully build the image. The resulting image has libcap 2.69, so that should be all that's necessary, as long as the alpine version upgrade doesn't cause any additional problems.

I didn't try installing 2.66-r1 on the older alpine image yet, but I can try that if it's less likely to cause some other problems.

(Filing this here because linkerd2-proxy doesn't have a /discussions page)

**Comment by risingspiral** [comment, marked as answer]
We've merged a PR to address this: https://github.com/linkerd/linkerd2-proxy-init/pull/269 it will go out in the next Linkerd edge and stable releases.
